lib/govuk_publishing_components/app_helpers/asset_helper.rb

```diff
@@ -0,0 +1,155 @@
+module GovukPublishingComponents
+  module AppHelpers
+    class AssetHelper
+      # This is a list of all the components that have stylesheets - so a new
+      # component needs to be added to this list rather than being laboriously
+      # added and removed from every single rendering application.
+      COMPONENT_CSS_PATHS = %w[
+        accordion
+        action-link
+        attachment
+        attachment-link
+        back-link
+        big-number
+        breadcrumbs
+        button
+        cards
+        character-count
+        checkboxes
+        contents-list
+        contextual-guidance
+        contextual-sidebar
+        cookie-banner
+        copy-to-clipboard
+        date-input
+        details
+        devolved-nations
+        document-list
+        emergency-banner
+        error-alert
+        error-message
+        error-summary
+        feedback
+        fieldset
+        file-upload
+        govspeak
+        govspeak-html-publication
+        heading
+        hint
+        image-card
+        input
+        inset-text
+        intervention
+        inverse-header
+        label
+        layout-footer
+        layout-for-admin
+        layout-for-public
+        layout-header
+        layout-super-navigation-header
+        lead-paragraph
+        metadata
+        modal-dialogue
+        notice
+        organisation-logo
+        panel
+        phase-banner
+        previous-and-next-navigation
+        print-link
+        radio
+        related-navigation
+        reorderable-list
+        search
+        select
+        share-links
+        show-password
+        signup-link
+        single-page-notification-button
+        skip-link
+        step-by-step-nav
+        step-by-step-nav-header
+        step-by-step-nav-related
+        subscription-links
+        success-alert
+        summary-list
+        table
+        tabs
+        textarea
+        title
+        translation-nav
+        warning-text
+      ].map { |component|
+        "govuk_publishing_components/components/_#{component}.css"
+      }.freeze
+
+      def initialize
+        # Set up the lists of gem components, app components, and views.
+        @gem_component_stylesheets_being_used = []
+        @app_component_stylesheets_being_used = []
+        @view_stylesheets_being_used = []
+
+        # This list includes components already included in Static; taken from
+        # https://github.com/alphagov/static/blob/198a598682df40ce4a2c3c286c06244297c18cf0/app/assets/stylesheets/application.scss
+        #
+        # This is used to dedupe stylesheets without needing the full path.
+        @stylesheet_list = %w[
```

Comment:
It'd be more consistent for this to be flagged as a class constant.

Reply:
`@stylesheet_list` should contain a unique list of component stylesheets that are required for the page being rendered. It's used to prevent the same stylesheet being requested multiple times, several buttons on a page, for example.

In the example above, we are pre-populating the `@stylesheet_list` with the stylesheets we know are already included in static. If a stylesheet is missed, it would lead to duplicate code being sent to the frontend, for example `button.css` is included in `static`, we would then add `button.css` again for the `frontend` rendering app.